app/cluster/state/static.go

```diff
@@ -1,29 +1,44 @@
 package state
 
 import (
+	"context"
+
 	"github.com/rudderlabs/rudder-server/app/cluster"
 	"github.com/rudderlabs/rudder-server/utils/types/servermode"
-	"sync"
+	"github.com/rudderlabs/rudder-server/utils/types/workspace"
 )
 
 var _ cluster.ModeProvider = &StaticProvider{}
 
 type StaticProvider struct {
-	Mode   servermode.Mode
-	ch     chan servermode.Ack
-	chOnce sync.Once
+	Mode servermode.Mode
+	// TODO: WorkspaceIDs
 }
 
-func (s *StaticProvider) ServerMode() (<-chan servermode.Ack, error) {
-	s.chOnce.Do(func() {
-		s.ch = make(chan servermode.Ack, 1)
+// ServerMode returns a channel with a single server mode, the one provided in the constructor.
+func (s *StaticProvider) ServerMode(ctx context.Context) <-chan servermode.ModeRequest {
+	ch := make(chan servermode.ModeRequest, 1)
+	ch <- servermode.NewModeRequest(servermode.Mode(s.Mode), func() error {
+		return nil
 	})
-	s.ch <- servermode.WithACK(servermode.Mode(s.Mode), func() {})
-	return s.ch, nil
+
+	go func() {
+		<-ctx.Done()
+		close(ch)
```

Comment:
Correct me if I'm wrong but nobody else can push on `ch` (given the return type) and we are pushing a single message only in it, correct?

If that stands then I got two observations:
1. there is no need for the routine to close the channel given that it's buffered, you could just push into it, close it and return it. there is no use for a context as well so it could be renamed `_ context.Context` because pushing a single message there won't be a blocking operation anyway.
2. unless the channel with one element is needed for some reason, couldn't we just return the element without the channel at all?

Reply:
The expected behaviour for the provider is to keep the channel open until context is cancelled.